libraries/Zigbee/src/ZigbeeEP.cpp

```diff
@@ -40,9 +42,22 @@ bool ZigbeeEP::setManufacturerAndModel(const char *name, const char *model) {
     log_e("Manufacturer or model name is too long");
     return false;
   }
+  // Get and check the basic cluster
+  esp_zb_attribute_list_t *basic_cluster = esp_zb_cluster_list_get_cluster(_cluster_list, ESP_ZB_ZCL_CLUSTER_ID_BASIC, ESP_ZB_ZCL_CLUSTER_SERVER_ROLE);
+  if (basic_cluster == nullptr) {
+    log_e("Failed to get basic cluster");
+    return false;
+  }  
   // Allocate a new array of size length + 2 (1 for the length, 1 for null terminator)
-  char *zb_name = new char[name_length + 2];
-  char *zb_model = new char[model_length + 2];
+  char *zb_name = (char *) malloc(name_length + 2);
+  char *zb_model = (char *) malloc(model_length + 2);
+  if (zb_name == nullptr || zb_model == nullptr) {
```

Comment:
nitpick: You are using "C" AP (`malloc`)I. You should check against C value (`NULL`) not "C++" (`nullptr`). Same for the rest of the code

Reply:
I'll change it back to new/delete.

Reply:
no. please leave it as malloc/free, just check against NULL for posterity

Reply:
Changed it to use stack alocated memory instead. It is 68 bytes on stack.
It is faster and there is no heap memory fragmentation side effect.